src/value/ser.rs

```diff
@@ -517,12 +521,20 @@ impl serde::Serializer for MapKeySerializer {
         Ok(value.to_string())
     }
 
-    fn serialize_f32(self, _value: f32) -> Result<String> {
-        Err(key_must_be_a_string())
+    fn serialize_f32(self, value: f32) -> Result<String> {
+        if value.is_finite() {
+            Ok(value.to_string())
+        } else {
+            Err(float_key_must_be_finite())
+        }
     }
 
-    fn serialize_f64(self, _value: f64) -> Result<String> {
-        Err(key_must_be_a_string())
+    fn serialize_f64(self, value: f64) -> Result<String> {
+        if value.is_finite() {
+            Ok(value.to_string())
```

Comment:
Please use the same logic used elsewhere in the crate for printing floats. A `Map<NotNan<f64>, NotNan<f64>>` should always produce the same digits in the keys and values. As written here, this uses std::fmt's representation, which is different from what we do in JSON. For example in the case of 1e-20.

Reply:
I've now updated this to use `CompactFormatter` with a `Vec<u8>` buffer, to keep formatting methods consistent. Would it make more sense to directly use `ryu` here?

Reply:
It would make more sense to directly use ryu.

Reply:
Updated to use `ryu`.